### PR TITLE
General refactor of bot.py, log.py and pagination.py

### PR DESCRIPTION
**Pull request by nerd-bear**
There are a few small changes that I made thought these files.
For the log file:

> Broken down the setup function to be more understandable and readable.
> Improved name of the _set_trace_loggers function and it now uses a single underscore, which is more conventional for internal use.
> Added an early return in the _set_trace_loggers function. Now a early return is called if level_filter is false, making it more readable.
> This goes for all the files, added consistent spacing and indentation to everything.
> Also added a bit more type hits for new contributors to better understand the code base. 

For the bot file:
 
> Switched "from sys import exception" with "from sys import exc_info", because the exc_info() function gives more detailed information.
> Changed new_scope to push_scope, because its the more used and recommended method.
> Changed the __init__ method to include the base exceptions message in the new exception.
> Removed empty __init__ method from Bot class as it was just calling the superclass constructor without any changes.
> Improved the while loop and made it a for loop with a range, making it more 'Pythonic' if you will and easier to read.

And finally for the pagination.py file:

> Both the class and the paginate method now have more info filled docstrings that explain their meaning and usage.
> The parameters in the paginate method signature are now aligned for better readability.
> The custom PaginationEmojis instance is now created within the method body, making it easier to modify if needed in the future.
> The parameters are passed directly to the superclass method, reducing redundancy and improving maintainability.

I hope this passes the style guide lines and I hope this change is not to small to be ignored!

### CODE REVIEW

File: bot/bot.py [hunk 1]
```diff
@@ -57,23 +44,18 @@ async def setup_hook(self) -> None:
...
         """Log errors raised in event listeners rather than printing them to stderr."""
-        e_val = exception()
-
-        if isinstance(e_val, Forbidden):
+        _, error, _ = exc_info()
+        if isinstance(error, Forbidden):
             message = args[0] if event == "on_message" else args[1] if event == "on_message_edit" else None
-
             with contextlib.suppress(Forbidden):
-                # Attempt to handle the error. This reraises the error if's not due to a block,
```
Comment (shtlrs) [inline bot/bot.py:66]: I guess you could leave the comment.

File: bot/log.py [hunk 2]
```diff
@@ -63,29 +38,50 @@ def setup_sentry() -> None:
...
+
+    if level_filter.startswith("*"):
+        get_logger().setLevel(core_logging.TRACE_LEVEL)
+    elif level_filter.startswith("!"):
+        get_logger().setLevel(core_logging.TRACE_LEVEL)
+        for logger_name in level_filter.strip("!,").split(","):
+            get_logger(logger_name).setLevel(logging.DEBUG)
+    else:
+        for logger_name in level_filter.strip(",").split(","):
+            get_logger(logger_name).setLevel(core_logging.TRACE_LEVEL)
```
Comment (shtlrs) [inline bot/log.py:87]: I thought ruff would complain once you have execution paths that do not return none when you have at least one that does. surprising.

**Comment by n0Oo0Oo0b** [comment]
We appreciate your effort into this PR, but there are some issues with the changes you've proposed here:

---

> Broken down the setup function to be more understandable and readable.

I don't see the benefit of this, especially considering the docstring of `setup()` already indicates that it's meant to "set up loggers". If you find readability to be an issue in the existing code (IMO it's fine as it is), labeling each section of the function with a comment would be more reasonable.

> Improved name of the `_set_trace_loggers` function and it now uses a single underscore, which is more conventional for internal use.

I'm not sure what change you're referring to, this function was (and still is) prefixed with a single underscore.

> Added an early return in the _set_trace_loggers function. Now a early return is called if level_filter is false, making it more readable.

I'm fine with this change, but one might argue that the existing code would be easier to work with if something else needed to be added in the future.

> This goes for all the files, added consistent spacing and indentation to everything.

Consistent with what? We use `ruff` to format code in our repositories, so the formatting was already perfectly consistent before. Changing the formatting broke this, which is why merging of this PR is currently blocked.

> Also added a bit more type hits for new contributors to better understand the code base.

Again, I'm not sure what change this is referring to.

> Switched "from sys import exception" with "from sys import exc_info", because the exc_info() function gives more detailed information.

The additional info isn't needed in this case, so it seems pretty pointless to change this.

> Changed new_scope to push_scope, because its the more used and recommended method.

According to the [sentry docs](https://docs.sentry.io/platforms/python/enriching-events/scopes/#changing-the-scope), `push_scope` looks like it's for an older version so it shouldn't even exist anymore.

> Removed empty init method from Bot class as it was just calling the superclass constructor without any changes.

Looks alright, I don't have any objections on this.

> Improved the while loop and made it a for loop with a range, making it more 'Pythonic' if you will and easier to read.

Looks okay to me, but there might have been some reason it was like that before. A third opinion on this would be nice.

> Both the class and the paginate method now have more info filled docstrings that explain their meaning and usage.

I think the existing docstring deferring to the superclass is better since it's one less change if that API is changed down the line. Also note that function docstrings should complete the sentence "This function/method ..." (i.e. should start with a verb).

> The parameters in the paginate method signature are now aligned for better readability.

I'm okay with this as long as ruff is happy with it.

> The custom PaginationEmojis instance is now created within the method body, making it easier to modify if needed in the future.

This could easily be changed later if required, I'm not sure why this needs to be changed now, especially for something as simple as this.

>  The parameters are passed directly to the superclass method, reducing redundancy and improving maintainability.

Assuming by this you mean forwarding `**kwargs` to `super().paginate()`, I don't mind this, although there isn't a clear need for it.

---

If you wish to continue contributing to our projects, I would recommend re-reading our [contributing guidelines](https://www.pythondiscord.com/pages/guides/pydis-guides/contributing/contributing-guidelines/) to ensure your PRs are up to standard and are heading in the right direction.

File: bot/pagination.py [hunk 3]
```diff
@@ -28,20 +24,26 @@ async def paginate(
...
 
-        Acts as a wrapper for the super class' `paginate` method to provide the pagination emojis by default.
+        This method wraps the superclass's `paginate` method, providing custom pagination emojis by default.
+        All parameters are passed directly to the superclass method.
 
-        Consult the super class's `paginate` method for detailed information.
+        Returns:
+            discord.Message | None: The message object if pagination is successful, None otherwise.
         """
+        pagination_emojis = PaginationEmojis(delete=Emojis.trashcan)
```
Comment (shtlrs) [inline bot/pagination.py:43]: You don't really need the variable if it's only being used to pass it as argument to the `pagination_emojis` param.

File: bot/bot.py [hunk 4]
```diff
@@ -1,52 +1,39 @@
...
+                log.info(f"Attempting site connection: {attempt}/{constants.URLs.connect_max_retries}")
                 await self.api_client.get("healthcheck")
-                break
-
+                return
             except (aiohttp.ClientConnectorError, aiohttp.ServerDisconnectedError):
-                attempts += 1
-                if attempts == constants.URLs.connect_max_retries:
+                if attempt == constants.URLs.connect_max_retries:
                     raise
```
Comment (shtlrs) [inline bot/bot.py:37]: We can make use of the `StartupError` exception here and raise it instead of a bare exception.

**Review by shtlrs — COMMENTED** [review]
Thanks for the changes, couple of small comments.

File: bot/log.py (hunk 2 above)
Comment (etrotta) [reply bot/log.py:87]: All paths in that function return None, just implicitly (one from a bare return statement, one from having no explicit return statement)

Comment (shtlrs) [reply bot/log.py:87]: > All paths in that function return None, just implicitly 

Yes that i know, it's just that i thought ruff would either impose that all returns be implicit, and when you have at leat one explicit, the rest need to become as such as well

**Pull request closed without merging**